app/backend/src/tests/test_events.py

```diff
@@ -0,0 +1,1726 @@
+        assert to_aware_datetime(res.end_time) == end_time
+        # assert res.timezone == "UTC"
+        assert res.start_time_display
+        assert res.end_time_display
+        assert res.attendance_state == events_pb2.ATTENDANCE_STATE_NOT_GOING
+        assert not res.organizer
+        assert not res.subscriber
+        assert res.going_count == 1
+        assert res.maybe_count == 0
+        assert res.organizer_count == 1
+        assert res.subscriber_count == 1
+        assert res.owner_user_id == user1.id
+        assert not res.owner_community_id
+        assert not res.owner_group_id
+        assert res.thread_id
+        assert not res.can_edit
+        assert res.can_moderate
+
+    with events_session(token3) as api:
+        res = api.GetEvent(events_pb2.GetEventReq(event_id=event_id))
+
+        assert res.is_next
+        assert not res.is_past
+        assert res.is_future
+        assert res.title == "Dummy Title"
+        assert res.slug == "dummy-title"
+        assert res.content == "Dummy content."
+        assert not res.photo_url
+        assert res.WhichOneof("mode") == "offline_information"
+        assert res.offline_information.lat == 0.1
+        assert res.offline_information.lng == 0.2
+        assert res.offline_information.address == "Near Null Island"
+        assert time_before < to_aware_datetime(res.created) < now()
+        assert time_before < to_aware_datetime(res.last_edited) < now()
+        assert res.creator_user_id == user1.id
+        assert to_aware_datetime(res.start_time) == start_time
+        assert to_aware_datetime(res.end_time) == end_time
+        # assert res.timezone == "UTC"
+        assert res.start_time_display
+        assert res.end_time_display
+        assert res.attendance_state == events_pb2.ATTENDANCE_STATE_NOT_GOING
+        assert not res.organizer
+        assert not res.subscriber
+        assert res.going_count == 1
+        assert res.maybe_count == 0
+        assert res.organizer_count == 1
+        assert res.subscriber_count == 1
+        assert res.owner_user_id == user1.id
+        assert not res.owner_community_id
+        assert not res.owner_group_id
+        assert res.thread_id
+        assert not res.can_edit
+        assert not res.can_moderate
+
+    with events_session(token1) as api:
+        # online only event
+        res = api.CreateEvent(
+            events_pb2.CreateEventReq(
+                title="Dummy Title",
+                content="Dummy content.",
+                photo_key=None,
+                online_information=events_pb2.OnlineEventInformation(
+                    link="https://app.couchers.org/meet/",
+                ),
+                parent_community_id=c_id,
+                start_time=Timestamp_from_datetime(start_time),
+                end_time=Timestamp_from_datetime(end_time),
+                timezone="UTC",
+            )
+        )
+
+        assert res.is_next
```

Comment:
Can two simultaneous events both be "next"?

Reply:
No. I've thought about it, and I think occurences of the same event should not overlap.